File: py/utility/make_error.py
# ZefoyTiktokBot SDK utility: make_error

from __future__ import annotations
from core.operation import ZefoyTiktokBotOperation
from core.result import ZefoyTiktokBotResult
from core.control import ZefoyTiktokBotControl
from core.error import ZefoyTiktokBotError


def make_error_util(ctx, err):
    if ctx is None:
        from core.context import ZefoyTiktokBotContext
        ctx = ZefoyTiktokBotContext({}, None)

    op = ctx.op
    if op is None:
        op = ZefoyTiktokBotOperation({})
    opname = op.name
    if opname == "" or opname == "_":
        opname = "unknown operation"

    result = ctx.result
    if result is None:
        result = ZefoyTiktokBotResult({})
    result.ok = False

    if err is None:
        err = result.err
    if err is None:
        err = ctx.make_error("unknown", "unknown error")

    errmsg = ""
    if isinstance(err, ZefoyTiktokBotError):
        errmsg = err.msg
    elif hasattr(err, "msg") and err.msg is not None:
        errmsg = err.msg
    elif isinstance(err, str):
        errmsg = err
    else:
        errmsg = str(err)

    msg = "ZefoyTiktokBotSDK: " + opname + ": " + errmsg
    msg = ctx.utility.clean(ctx, msg)

    result.err = None

    spec = ctx.spec

    if ctx.ctrl.explain is not None:
        ctx.ctrl.explain["err"] = {"message": msg}

    sdk_err = ZefoyTiktokBotError("", msg, ctx)
    sdk_err.result = ctx.utility.clean(ctx, result)
    sdk_err.spec = ctx.utility.clean(ctx, spec)

    if isinstance(err, ZefoyTiktokBotError):
        sdk_err.code = err.code

    ctx.ctrl.err = sdk_err

    if ctx.ctrl.throw_err is False:
        return result.resdata, None

    return None, sdk_err

File: .sdk/tm/php/utility/MakeError.php
<?php
declare(strict_types=1);

// ZefoyTiktokBot SDK utility: make_error

require_once __DIR__ . '/../core/Operation.php';
require_once __DIR__ . '/../core/Result.php';
require_once __DIR__ . '/../core/Error.php';

class ZefoyTiktokBotMakeError
{
    public static function call(?ZefoyTiktokBotContext $ctx, mixed $err): array
    {
        if ($ctx === null) {
            require_once __DIR__ . '/../core/Context.php';
            $ctx = new ZefoyTiktokBotContext([], null);
        }
        $op = $ctx->op ?? new ZefoyTiktokBotOperation([]);
        $opname = $op->name;
        if ($opname === '' || $opname === '_') {
            $opname = 'unknown operation';
        }

        $result = $ctx->result ?? new ZefoyTiktokBotResult([]);
        $result->ok = false;

        if ($err === null) {
            $err = $result->err;
        }
        if ($err === null) {
            $err = $ctx->make_error('unknown', 'unknown error');
        }

        $errmsg = ($err instanceof ZefoyTiktokBotError) ? $err->msg : (string)$err;
        $msg = "ZefoyTiktokBotSDK: {$opname}: {$errmsg}";
        $msg = ($ctx->utility->clean)($ctx, $msg);

        $result->err = null;
        $spec = $ctx->spec;

        if ($ctx->ctrl->explain) {
            $ctx->ctrl->explain['err'] = ['message' => $msg];
        }

        $sdk_err = new ZefoyTiktokBotError('', $msg, $ctx);
        $sdk_err->result = ($ctx->utility->clean)($ctx, $result);
        $sdk_err->spec = ($ctx->utility->clean)($ctx, $spec);
        if ($err instanceof ZefoyTiktokBotError) {
            $sdk_err->sdk_code = $err->sdk_code;
        }

        $ctx->ctrl->err = $sdk_err;

        if ($ctx->ctrl->throw_err === false) {
            return [$result->resdata, null];
        }
        return [null, $sdk_err];
    }
}
